api/v1alpha1/fio_types.go

```diff
@@ -20,6 +20,18 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// FioPodSpec contains the configuration for the benchmark pods
+type FioPodConfigurationSpec struct {
```

Comment:
Currently this deviates from iperf3 controller, there we have `PodLabels` and `PodScheduling` in the top level Spec.
I'm not against moving this under a common spec, but we should do this consistently throughout the whole project.

Now, given that we should also expose CPU and Memory Limits (and maybe others), it is a good idea to have a common struct as you propose. 

Can you please move this struct to `k8s_types.go` in api/ ?

CC: @otto-von-bivouac

Reply:
?
On the contrary, I'm using the same structure as in iperf3: https://github.com/xridge/kubestone/blob/258970513f9e89687c9ee0921c43b5699fbd0c47/api/v1alpha1/iperf3_types.go#L26

However, I do agree that we should move these to k8s_types. The problem is, the two benchmarks need slightly different structures (that's why I kept it separated for now). Maybe do the refactor in a follow up PR?

Reply:
I think we should only factor the common ones: `PodScheduling`, `PodLabels`. 
In other words, Iperf3ConfigurationSpec should also use a `PodConfigurationSpec`, which contains those fields.

I'm okay with a follow-up PR, but then please add a TODO line for it.

Reply:
Nvm, I did it in 5a3284d.

Reply:
thanks. it is still differing on API level (inlined object in iperf3), but I'll address it in a separate PR.